src/blocks/rsvp-form/edit.js

```diff
@@ -8,36 +8,55 @@ import {
 } from '@wordpress/block-editor';
 import { PanelBody, ToggleControl } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { useState, useEffect } from '@wordpress/element';
+import { useState, useEffect, useRef } from '@wordpress/element';
 
 /**
  * Internal dependencies.
  */
 import TEMPLATE from './template';
 
 const Edit = () => {
-	const blockProps = useBlockProps();
+	const blockRef = useRef( null );
+	const blockProps = useBlockProps( { ref: blockRef } );
 	const [ showMessage, setShowMessage ] = useState( false );
 
 	// Toggle visibility of success message blocks for preview.
 	useEffect( () => {
-		const messageElements = document.querySelectorAll(
-			'.gatherpress-rsvp-form-message',
-		);
-		messageElements.forEach( ( element ) => {
-			element.style.display = showMessage ? 'block' : 'none';
-		} );
-	}, [ showMessage ] );
+		const updateMessageVisibility = () => {
+			if ( ! blockRef.current ) {
+				return;
+			}
 
-	// Hide message blocks immediately on mount.
-	useEffect( () => {
-		const messageElements = document.querySelectorAll(
-			'.gatherpress-rsvp-form-message',
-		);
-		messageElements.forEach( ( element ) => {
-			element.style.display = 'none';
-		} );
-	}, [] );
+			const messageElements = blockRef.current.querySelectorAll(
+				'.gatherpress--rsvp-form-message',
+			);
+
+			messageElements.forEach( ( element ) => {
+				element.style.setProperty(
+					'display',
+					showMessage ? 'block' : 'none',
+					'important',
+				);
```

Comment:
Using the css display property is by far not accessible, to screen reader users. Considering changing aria roles for the visibility state.